File: app/src/main/java/com/past/music/activity/CreateSongListActivity.java
package com.past.music.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.widget.EditText;
import android.widget.TextView;

import com.jaeger.library.StatusBarUtil;
import com.past.music.Config.BaseConfig;
import com.past.music.MyApplication;
import com.past.music.pastmusic.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class CreateSongListActivity extends AppCompatActivity {

    String listName;
    String listInfo;

    @BindView(R.id.et_list_name)
    EditText nameInput;

    @BindView(R.id.et_list_info)
    EditText infoInput;

    @OnClick(R.id.back)
    void back() {
        finish();
    }

    @BindView(R.id.title)
    TextView mTitle;

    @OnClick(R.id.save)
    void save() {
        listName = nameInput.getText().toString();
        listInfo = infoInput.getText().toString();
        if (!TextUtils.isEmpty(listName)) {
            MyApplication.songListDBService.insert(listName, listInfo);
        }
        finish();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_create_song_list);
        setStatusBar();
        ButterKnife.bind(this);

        mTitle.setText("新建歌单");
        nameInput.setText("新建歌单");
        nameInput.setSelectAllOnFocus(true);
        nameInput.requestFocus();
    }

    protected void setStatusBar() {
        int mColor = getResources().getColor(R.color.colorPrimary);
        StatusBarUtil.setColor(this, mColor, BaseConfig.Alpha);
    }

    @Override
    public void finish() {
        super.finish();
        this.overridePendingTransition(R.anim.push_up_in, R.anim.push_down_out);
    }
}

File: app/src/main/java/com/se/music/subpage/mine/CreateSongListActivity.kt
package com.se.music.subpage.mine

import android.content.ContentValues
import android.os.Bundle
import android.view.*
import android.widget.EditText
import com.se.music.R
import com.se.music.common.ToolBarActivity
import com.se.music.provider.SongList
import com.se.music.utils.IdUtils
import java.util.*

/**
 *Author: gaojin
 *Time: 2018/5/13 下午5:49
 */

class CreateSongListActivity : ToolBarActivity() {

    companion object {
        val resultCode = IdUtils.generateLoaderId()
    }

    lateinit var nameInput: EditText
    lateinit var infoInput: EditText

    override fun createContentView(inflater: LayoutInflater, rootView: ViewGroup): View {
        return inflater.inflate(R.layout.activity_create_song_list, rootView, false)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        title = resources.getString(R.string.create_new_song_list)

        nameInput = findViewById(R.id.et_list_name)
        infoInput = findViewById(R.id.et_list_info)
        nameInput.setText(resources.getString(R.string.create_new_song_list))
        nameInput.setSelectAllOnFocus(true)
        nameInput.requestFocus()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        super.onCreateOptionsMenu(menu)
        menuInflater.inflate(R.menu.menu_save, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            R.id.menu_save -> save()
        }
        return super.onOptionsItemSelected(item)
    }

    fun save() {
        val listName = nameInput!!.text.toString()
        val listInfo = infoInput!!.text.toString()

        val values = ContentValues()
        values.put(SongList.ID, UUID.randomUUID().toString())
        values.put(SongList.NAME, listName)
        values.put(SongList.CREATE_TIME, System.currentTimeMillis())
        values.put(SongList.INFO, listInfo)
        contentResolver.insert(SongList.CONTENT_URI, values)
        setResult(resultCode)
        finish()
    }

    override fun finish() {
        super.finish()
        overridePendingTransition(R.anim.push_up_in, R.anim.push_down_out)
    }

}